### PR TITLE
Fix the output shape in python to match the C code for DownsampleFactorMaxGradGrad.

### PR DESCRIPTION
**Pull request by abergeron**
This should fix the DebugMode errors in the buildbot.

### CODE REVIEW

File: theano/tensor/signal/pool.py [hunk 1]
```diff
@@ -914,8 +914,8 @@ def perform(self, node, inp, out):
         if len(x.shape) != 4:
             raise NotImplementedError(
                 'DownsampleFactorMaxGradGrad requires 4D input for now')
-        if (z[0] is None) or (z[0].shape != x.shape):
```
Comment (nouiz) [inline theano/tensor/signal/pool.py]: Can you confirm that there was a test failure under debugmode?

Comment (nouiz) [inline theano/tensor/signal/pool.py]: If so, I'm good to merge when travis pass.

Comment (abergeron) [inline theano/tensor/signal/pool.py]: Yes it was.  And actually multiple test failed because of that.

**Comment by abergeron** [comment]
I've fixed another problem related to DownsampleFactorMax too and I had to rebase.

**Comment by abergeron** [comment]
That one also caused problems in DebugMode.

**Comment by nouiz** [comment]
travis is failing with this error:

Warning, treated as error:

/home/travis/build/Theano/Theano/doc/library/tensor/signal/downsample.txt:14: WARNING: autodoc: failed to import function u'max_pool_2d' from module u'theano.tensor.signal.downsample'; the following exception was raised:

Traceback (most recent call last):

  File "/home/travis/miniconda2/envs/pyenv/lib/python2.6/site-packages/Sphinx-1.3.1-py2.6.egg/sphinx/ext/autodoc.py", line 385, in import_object

    __import__(self.modname)

  File "/home/travis/build/Theano/Theano/theano/tensor/signal/downsample.py", line 14, in <module>

    local_average_pool_grad = pool.local_average_pool_grad

AttributeError: 'module' object has no attribute 'local_average_pool_grad'

**Comment by nouiz** [comment]
travis fail:

  File "/home/travis/build/Theano/Theano/theano/tensor/signal/tests/test_pool.py", line 11, in <module>

    from theano.tensor.signal.pool import (Pool, pool_2d,

ImportError: cannot import name DownsampleFactorMaxGrad

**Pull request merged**